README.md

```diff
@@ -1 +1,241 @@
+The `avr-libstdcpp` port (even though not-fully-tested
+and only partially full/complete) will, nonetheless,
+be useful for those interested in making more comprehensive utilization
+of C++ --- including its standard library --- with a modern `avr-gcc` compiler.
+
+## Historical origins
+
+The `avr-libstdcpp` library traces its own origins to existing
+GNU/GCC C++ standard library implementaion(s) --- themselves targeting
+embedded systems. This work is essentially an even-more
+_embedded_-_friendly_ adaptation of the aforementioned work.
+
+The `avr-libstdcpp` port began in 2018 with an initial import
+of a GNU/GCC-based C++ standard library from GCC 8.
+A second import of a GNU/GCC-based C++ standard library from GCC 10
+in 2020 modernized the port to include many modern C++20 features.
+
+## Using the library
+
+Add the `avr-libstdcpp/include` path to the standard `-I`
+include path(s) of the compiler on the command line.
+Upon doing this, include standard library headers in the usual way.
+
+For example:
+
+Simply add the `-I` include path to your particular location
+of `avr-libstdcpp/include` on the command line...
+
+```sh
+avr-g++ -O2 -x c++ -I/my_path/avr-libstdcpp/include -mmcu=atmega328p test.cpp -o test.elf
+```
+
+... and seamlessly use standard library headers in your code.
+
+```C
+#include <array>
+#include <numeric>
+
+std::array<int, 3U> v {{ 1, 2, 3 }};
+
+int main()
+{
+  // 6
+  auto sum = std::accumulate(v.cbegin(), v.cend(), 0);
+
+  static_cast<void>(sum);
+}
+```
+
+Additional straightforward code samples exercising standard library usage
+can be found in the [examples](./examples) folder.
+
+## Guidance for tiny bare-metal systems
+
+### Very helpful go-to libraries
+
+In general the C++ standard library is intended
+to be written and implemented in a resource-sensitive
+fashion. This includes efforts to save on both
+memory as well as run-time. Algorithms are written
+with efficiency in mind and memory footprints are
+generally kept as minimal as possible.
+
+Some library components, however, are particularly well-suited
+for bare-metal microcontroller programming.
+These can be exceptionally helpful when used properly and sensibly
+in tiny bare-metal microcontroller environments.
+
+A subjective list of these the libraries/headers
+and their main uses includes, but is not limited to,:
+
+- `<array>` for containers having known, fixed size.
+- `<algorithm>` for standard algorithms such as sorting, minimax, sequential operations, etc.
+- `<cmath>` for projects requiring floating-point mathematical functions such as `std::sin()`, `std::exp()`, `std::frexp()` and many more.
+- `<cstdint>` which defines integral types having specified widths residing within `namespace` `std` like `std::uint8_t`.
+- `<limits>` offering compile-time query of numeric limits of built-in types.
+- `<numeric>` featuring a collection of useful numeric algorithms such ass `std::accumulate()`, etc.
```

Comment:
```suggestion
- `<numeric>` featuring a collection of useful numeric algorithms such as `std::accumulate()`, etc.
```
butt

Reply:
Ugggh! Where is the spell checker when you need it?